### ISSUE TITLE
When build mxnet from source via cmake, if I set `USE_SSE` to `0`, target `mshadow` is still built with SSE  option on. 

### ISSUE THREAD

**Issue by TsingWei** [apache/incubator-mxnet#16809]
## Description
When build mxnet from source via cmake, if I set `USE_SSE` to `0`, target `mshadow` is still built with SSE  option on. 

### Error Message
```bash
[  3%] Building CXX object CMakeFiles/mxnet_static.dir/src/c_api/c_api.cc.o
/usr/bin/c++  -DDMLC_USE_CXX11 -DDMLC_USE_CXX11=1 -DMSHADOW_INT64_TENSOR_SIZE=0 -DMSHADOW_IN_CXX11 -DMSHADOW_USE_CBLAS=1 -DMSHADOW_USE_CUDA=1 -DMSHADOW_USE_CUDNN=1 -DMSHADOW_USE_F16C=0 -DMSHADOW_USE_MKL=0 -DMXNET_ENABLE_CUDA_RTC=1 -DMXNET_USE_BLAS_OPEN=1 -DMXNET_USE_CUDA=1 -DMXNET_USE_LAPACK=1 -DMXNET_USE_NCCL=0 -DMXNET_USE_NVTX=1 -DMXNET_USE_OPENCV=1 -DMXNET_USE_OPENMP=1 -DMXNET_USE_OPERATOR_TUNING=1 -DMXNET_USE_SIGNAL_HANDLER=1 -DMXNET_USE_TENSORRT=1 -DNDEBUG=1 -DONNX_NAMESPACE=onnx -DUSE_CUDNN -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/onnx-tensorrt/third_party/onnx -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/onnx-tensorrt -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/include -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/src -I/usr/local/cuda/include -I/usr/local/cuda/targets/aarch64-linux/include -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/mshadow -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/nvidia_cub -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/tvm/nnvm/include -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/tvm/include -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/dmlc-core/include -I/home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/dlpack/include -isystem /usr/include/opencv  -Wall -Wno-unknown-pragmas -Wno-sign-compare -O3 -std=c++11 -fopenmp -std=c++0x -fPIC   -o CMakeFiles/mxnet_static.dir/src/c_api/c_api.cc.o -c /home/nvidia/apache-mxnet-src-1.5.1-incubating/src/c_api/c_api.cc
In file included from /home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/mshadow/mshadow/././././extension/../packet-inl.h:203:0,
                 from /home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/mshadow/mshadow/././././extension/implicit_gemm.h:11,
                 from /home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/mshadow/mshadow/./././dot_engine-inl.h:12,
                 from /home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/mshadow/mshadow/././expr_engine-inl.h:431,
                 from /home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/mshadow/mshadow/./extension.h:10,
                 from /home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/mshadow/mshadow/tensor.h:1054,
                 from /home/nvidia/apache-mxnet-src-1.5.1-incubating/include/mxnet/base.h:33,
                 from /home/nvidia/apache-mxnet-src-1.5.1-incubating/src/c_api/c_api.cc:38:
/home/nvidia/apache-mxnet-src-1.5.1-incubating/3rdparty/mshadow/mshadow/././././extension/../packet/sse-inl.h:10:10: fatal error: emmintrin.h: No such file or directory
 #include <emmintrin.h>
```

## To Reproduce
Prepare s device without SSE support(for me it's TX2 which is  aarch) and run cmake with `USE_SSE` option off.

## What have you tried to solve it?

1. add `add_definitions(-DMSHADOW_USE_SSE=0)` after line 135 https://github.com/apache/incubator-mxnet/blob/017f6fa7996048cc429b02dac9369af6778811ec/CMakeLists.txt#L135

It works but it is just a simple way.

## Environment

nvidia jetson TX2, ubuntu 18.04

**Comment by lanking520** [comment]
@szha

**Comment by TsingWei** [comment]
It was fixed in #17031 .
